### DISCUSSION TITLE
q-page-container and keep-alive, doesn't recalculate q-header height when switching route.

### DISCUSSION THREAD

**Discussion by altezza04** [quasarframework/quasar#16647] [General - Components / Directives / etc]
When using keep-alive inside q-page-container, the heading seem to always cover the q-page after switching route. it seem it doesn't recalculate the q-header height in keep-alive, any way to 'force' q-page-container to recalculate the padding-top value?

`   <q-page-container>
      <router-view v-slot="{ Component }">
        <keep-alive>
          <component :is="Component" />
        </keep-alive>
      </router-view>`

**Comment by pdanpdan** [comment]
As far as I remember the solution from Vue was to give a key to the component (you can try the url or part of it)

**Comment by altezza04** [comment]
I did try with :key="route.path" or :key="route.fullPath" still not working. But if I try to resize the broweser window, the q-header height will recalculate property

**Comment by weberdepaulai12** [comment]
Did you find any solution to this problem? 

I'm having the same problem

**Reply by altezza04** [reply]
I just use this 

`    <q-page-container :style="{ paddingTop: $route.meta.layoutPaddingTop}">
`

and set the padding-top of the page manually with router meta
